### PR TITLE
Make the code more PowerShell compliant

### LINKED ISSUE #39463: The PowerShell Ansible.ModuleUtils.Legacy.psm1 is not fully PowerShell compliant
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME
Ansible.ModuleUtils.Legacy.psm1
##### ANSIBLE VERSION
```
ansible 2.5.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/wojtek/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.14 (default, Mar 14 2018, 13:36:31) [GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
```

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### SUMMARY
The code in the file [Ansible.ModuleUtils.Legacy.psm1](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/powershell/Ansible.ModuleUtils.Legacy.psm1) isn't PowerShell compliant. The PowerShell PSScriptAnalyzer v. 1.6.1 (that is PEP8-like for PowerShell) reports warnings that can be easy corrected.

For reported warnings
- PSAvoidUsingCmdletAliases  - the code readability will be improved
- PSPossibleIncorrectComparisonWithNull - possible unexpected behaviors under execution can be mitigated

Additionally, existing comments intended to be general help are not PowerShell compliant as the result Get-Help can't be used to check the module content.

##### STEPS TO REPRODUCE

For the PSScriptAnalyzer warnings
```
Import-Module PSScriptAnalyzer

Invoke-Script Invoke-ScriptAnalyzer ./Ansible.ModuleUtils.Legacy.psm1 

\<Output is in the expected results section below\>

```

For the Get-Help
```
Import-Module ./Ansible.ModuleUtils.Legacy.psm1 

Get-Command -Module Ansible.ModuleUtils.Legacy | ForEach-Object { Get-Help $_.Name }
```
doesn't returns any constructive explenations for commands.

##### EXPECTED RESULTS
- Fewer warnings for the PSScriptAnalyzer analysis results.
- Get-Help displayed in the PowerShell way.

##### ACTUAL RESULTS
```
RuleName                            Severity     ScriptName Line  Message
--------                            --------     ---------- ----  -------
PSAvoidUsingCmdletAliases           Warning      Ansible.Mo 44    'echo' is an alias of 'Write-Output'. Alias can introduce
                                                 duleUtils.       possible problems and make scripts hard to maintain. Please
                                                 Legacy.psm       consider changing alias to its full content.
                                                 1
PSAvoidUsingCmdletAliases           Warning      Ansible.Mo 74    'echo' is an alias of 'Write-Output'. Alias can introduce
                                                 duleUtils.       possible problems and make scripts hard to maintain. Please
                                                 Legacy.psm       consider changing alias to its full content.
                                                 1
PSAvoidUsingCmdletAliases           Warning      Ansible.Mo 323   'where' is an alias of 'Where-Object'. Alias can introduce
                                                 duleUtils.       possible problems and make scripts hard to maintain. Please
                                                 Legacy.psm       consider changing alias to its full content.
                                                 1
PSAvoidUsingWMICmdlet               Warning      Ansible.Mo 321   File 'Ansible.ModuleUtils.Legacy.psm1' uses WMI cmdlet. For
                                                 duleUtils.       PowerShell 3.0 and above, use CIM cmdlet which perform the
                                                 Legacy.psm       same tasks as the WMI cmdlets. The CIM cmdlets comply with
                                                 1                WS-Management (WSMan) standards and with the Common
                                                                  Information Model (CIM) standard, which enables the cmdlets
                                                                  to use the same techniques to manage Windows computers and
                                                                  those running other operating systems.
PSPossibleIncorrectComparisonWithNu Warning      Ansible.Mo 55    $null should be on the left side of equality comparisons.
ll                                               duleUtils.
                                                 Legacy.psm
                                                 1
PSPossibleIncorrectComparisonWithNu Warning      Ansible.Mo 113   $null should be on the left side of equality comparisons.
ll                                               duleUtils.
                                                 Legacy.psm
                                                 1
PSPossibleIncorrectComparisonWithNu Warning      Ansible.Mo 144   $null should be on the left side of equality comparisons.
ll                                               duleUtils.
                                                 Legacy.psm
                                                 1
PSPossibleIncorrectComparisonWithNu Warning      Ansible.Mo 154   $null should be on the left side of equality comparisons.
ll                                               duleUtils.
                                                 Legacy.psm
                                                 1
PSPossibleIncorrectComparisonWithNu Warning      Ansible.Mo 181   $null should be on the left side of equality comparisons.
ll                                               duleUtils.
                                                 Legacy.psm
                                                 1
PSUseApprovedVerbs                  Warning      Ansible.Mo 51    The cmdlet 'Fail-Json' uses an unapproved verb.
                                                 duleUtils.
                                                 Legacy.psm
                                                 1
PSUseApprovedVerbs                  Warning      Ansible.Mo 259   The cmdlet 'Parse-Args' uses an unapproved verb.
                                                 duleUtils.
                                                 Legacy.psm
                                                 1
PSUseShouldProcessForStateChangingF Warning      Ansible.Mo 11    Function ’Set-Attr’ has verb that could change system
unctions                                         duleUtils.       state. Therefore, the function has to support
                                                 Legacy.psm 'ShouldProcess'.
                                                 1

```

### PR DESCRIPTION
**Pull request by it-praktyk**
##### SUMMARY
Make the code more PowerShell compliant
- decrease an amount of warning displayed by PSScriptAnalyzer (that is PEP8-like for PowerShell)
- improve displaying of help under PowerShell (existing comments are transformed to PowerShell formatted help)

Fixes #39463 

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME

Ansible.ModuleUtils.Legacy.psm1

##### ANSIBLE VERSION
```
ansible 2.5.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/wojtek/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.14 (default, Mar 14 2018, 13:36:31) [GCC 7.3.1 20180303 (Red Hat 7.3.1-5)]
```

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/63108/21/console

```
2018-04-28 12:13:27 windows_2012-R2 | FAILED! => {
2018-04-28 12:13:27     "changed": false, 
2018-04-28 12:13:27     "module_stderr": "ConvertTo-Json : Cannot bind parameter 'Depth'. Cannot convert value \"99#\" to \r\ntype \"System.Int32\". Error: \"Input string was not in a correct format.\"\r\nAt line:56 char:57\r\n+     Write-Output $obj | ConvertTo-Json -Compress -Depth 99# Example: \r\nExit-Json $ ...\r\n+                                                         ~~~\r\n    + CategoryInfo          : InvalidArgument: (:) [ConvertTo-Json], ParentCon \r\n   tainsErrorRecordException\r\n    + FullyQualifiedErrorId : CannotConvertArgumentNoMessage,Microsoft.PowerSh \r\n   ell.Commands.ConvertToJsonCommand\r\n \r\n\r\n", 
2018-04-28 12:13:27     "module_stdout": "", 
2018-04-28 12:13:27     "msg": "MODULE FAILURE", 
2018-04-28 12:13:27     "rc": 1
2018-04-28 12:13:27 }
```

**Comment by it-praktyk** [comment]
@jborean93 , @nitzmahone, can you review the change proposed by me? The CI build failed because the one exception (previously added) for PSScriptAnalyzer is not needed after that.

**Comment by it-praktyk** [comment]
@nitzmahone , @jborean93,
[failed tests](https://app.shippable.com/github/ansible/ansible/runs/73697/summary/console) IMHO are not related to the proposed change.

**Review by nitzmahone — APPROVED** [review]

File: lib/ansible/module_utils/powershell/Ansible.ModuleUtils.Legacy.psm1 [hunk 1]
```diff
@@ -303,7 +347,7 @@ Function Get-FileChecksum($path, $algorithm = 'sha1')
             $fp.Dispose();
         }
     }
-    ElseIf (Test-Path -Path $path -PathType Container)
+    Elseif (Test-Path -Path $path -PathType Container)
```
Comment (dagwieers) [inline lib/ansible/module_utils/powershell/Ansible.ModuleUtils.Legacy.psm1]: Minor nitpick, unless we standardize this, I would leave it as is.
This file either used `elseif` or `ElseIf`, introducing now also `Elseif` is adding to the entropy ;-)

Comment (it-praktyk) [reply lib/ansible/module_utils/powershell/Ansible.ModuleUtils.Legacy.psm1]: OK, I've decreased an entropy ;-)

**Review by dagwieers — APPROVED** [review]

**Review by jborean93 — APPROVED** [review]

**Pull request merged**